File: 038-数字在排序数组中出现的次数/038-GetNumberOfK.py
# -*- coding:utf-8 -*-
class Solution:
    def GetNumberOfK(self, data, k):
        # write code here
        if len(data)==0:
            return 0
        return self.bisect(data,k+0.5)-self.bisect(data,k-0.5)
    def bisect(self,data,k):
        left=0
        right=len(data)-1
        while left<=right:
            mid=(left+right)//2
            if data[mid]==k:
                return mid
            elif data[mid]<k:
                left=mid+1
            else:
                right=mid-1
        return left

File: 038-数字在排序数组中出现的次数/038-GetNumberOfK.cpp
class Solution {
public:
    int GetNumberOfK(vector<int> data ,int k) {
        if(data.size()==0){
            return 0;
        }
        return bisect(data,k+0.5)-bisect(data,k-0.5);
    }
    int bisect(vector<int> data,float k){
        int left=0;
        int right=data.size()-1;
        while (left<=right){
            int mid=(left+right)/2;
            if(data[mid]<k){
                left=mid+1;
            }
            else{
                right=mid-1;
            }
        }
        return left;
    }
};
